server/start.go

```diff
@@ -363,17 +365,39 @@ func startInProcess(ctx *server.Context, clientCtx client.Context, appCreator ty
 		}
 	}
 
-	var apiSrv *api.Server
-	if config.API.Enable {
+	if config.API.Enable || config.JSONRPC.Enable {
 		genDoc, err := genDocProvider()
 		if err != nil {
 			return err
 		}
 
-		clientCtx := clientCtx.
+		clientCtx = clientCtx.
 			WithHomeDir(home).
 			WithChainID(genDoc.ChainID)
 
+		// Set `GRPCClient` to `clientCtx` to enjoy concurrent grpc query.
+		if config.GRPC.Enable {
```

Comment:
shouldn't this block be added to L422 instead together with the other `if config.GRPC.Enable`?

Reply:
The grpc client is only needed by the grpc-gateway or our json-rpc service.

Reply:
The `config.GRPC.Enable` enables the server. It would be helpful to add more comments for extra context

Reply:
```suggestion
		// only use it if gRPC server is enabled.
		if config.GRPC.Enable {
```